core/src/main/scala/org/apache/spark/sql/crossdata/XDContext.scala

```diff
@@ -79,22 +79,28 @@ class XDContext private (@transient val sc: SparkContext,
   catalogConfig = xdConfig.getConfig(CoreConfig.CatalogConfigKey)
 
   @transient
-  override protected[sql] lazy val catalog: XDCatalog = {
+  override protected[sql] lazy val catalog: XDCatalogWithPersistence = {
 
     import XDContext.{CaseSensitive, DerbyClass}
 
-    val catalogClass = if (catalogConfig.hasPath(XDContext.ClassConfigKey))
+    val externalCatalogName = if (catalogConfig.hasPath(XDContext.ClassConfigKey))
       catalogConfig.getString(XDContext.ClassConfigKey)
     else DerbyClass
 
-    val xdCatalog = Class.forName(catalogClass)
+    val externalCatalogClass = Class.forName(externalCatalogName)
 
     val caseSensitive: Boolean = catalogConfig.getBoolean(CaseSensitive)
 
-    val constr: Constructor[_] = xdCatalog.getConstructor(classOf[CatalystConf], classOf[XDContext])
+    val constr: Constructor[_] = externalCatalogClass.getConstructor(classOf[CatalystConf], classOf[XDContext])
+
+    val conf = new SimpleCatalystConf(caseSensitive)
+
+    val externalCatalog = constr.newInstance(conf, self).asInstanceOf[PersistentCatalog]
+
+    val secondaryCatalogs = externalCatalog :: streamingCatalog.toList
```

Comment:
Although is not related to this PR, it would be nice to replace the logError launched when there is no streaming catalog with a logWarn/logInfo.